### ISSUE TITLE
CI: "Report environment details" step routinely times out after 1 minute, forcing workflow re-runs

### ISSUE THREAD

**Issue by etiotto** [intel/intel-xpu-backend-for-triton#6725]
## Summary

The `Report environment details` step in our CI workflows frequently hangs and is killed by its 1-minute `timeout-minutes` budget, failing the whole job and forcing a manual re-run. This happens often enough that it has become a regular "babysitting" tax on every CI run.

## Example failure

Observed on a recent run (branch `6693/merge`, TRITON_COMMIT_ID `9e2cab8db0a999f64db96f4240f689c859cf8fee`):

```
TIMESTAMP=20260421174816
GITHUB_RUN_ID=24737111888
GITHUB_RUN_NUMBER=17207
GITHUB_RUN_ATTEMPT=1
PYTHON_VERSION=3.10
PYTORCH_REPO=pytorch/pytorch
PYTORCH_COMMIT_ID=34d845da4749be49f76ba77480bb02a6a3779c4c
PYTORCH_VERSION=2.13.0a0+git34d845d
TRITON_REPO=intel/intel-xpu-backend-for-triton
TRITON_COMMIT_ID=9e2cab8db0a999f64db96f4240f689c859cf8fee
TRITON_BRANCH=6693/merge
OS=Linux
Error: The action 'Report environment details' has timed out after 1 minutes.
```

The log stops immediately after the last static env line (`OS=Linux`), i.e. after the heredoc `tee` completes but before (or during) `scripts/capture-hw-details.sh`.

## Where

The step is defined in multiple reusable workflows (all share the same shape):

- `.github/workflows/build-test-reusable.yml` (lines ~259–283)
- `.github/workflows/build-test-windows.yml`
- `.github/workflows/e2e-reusable.yml`
- `.github/workflows/e2e-windows.yml`
- `.github/workflows/inductor-tests-reusable.yml`
- `.github/workflows/nightly-wheels.yml`

Shape of the step:

```yaml
- name: Report environment details
  timeout-minutes: 1
  run: |
    ...
    cat <<EOF | tee "$reports_file"
    TIMESTAMP=...
    ...
    OS=${{ runner.os }}
    EOF
    ./scripts/capture-hw-details.sh | tee -a "$reports_file"
```

## Likely cause

Given that the log cuts off right as `capture-hw-details.sh` starts, the stall almost certainly comes from one of the external tools that script invokes:

- `clinfo --json` (can hang if the GPU stack / driver is in a bad state)
- `sycl-ls --verbose` (same concern)
- `dpkg-query --show libigc2 / libigc1 / libze1 / intel-level-zero-gpu`
- `python -c "import torch"` (can be slow if torch/CUDA/XPU init stalls)
- `icpx --version`

Any one of these stalling for >~30s will blow the 1-minute budget for the whole step.

## Impact

- Failing job is not a real product/test failure — it's pure CI noise.
- Every occurrence requires a manual re-run (triggering another full environment setup, checkout, build, etc.), which wastes CI capacity and developer attention.
- Masks real failures: a red "Report environment details" step distracts reviewers from genuinely failing jobs in the same run.

## Proposed mitigations (not mutually exclusive)

1. **Add per-command timeouts** in `scripts/capture-hw-details.sh` using `timeout <n>` around the slow probes (`clinfo`, `sycl-ls`, `python -c "import torch"`, `icpx --version`) so a single hung probe degrades to `Not Installed`/`Unknown` rather than taking the whole step down.
2. **Raise `timeout-minutes`** from `1` to e.g. `3` — the step's purpose is pure metadata capture; it doesn't need such an aggressive budget.
3. **Make the step non-fatal**: add `continue-on-error: true` so metadata-collection failures don't fail the whole job. The downstream reporting can tolerate missing fields.
4. **Log which command is slow**: wrap each probe with timestamped output so that when it does hang we can see which tool caused it (currently the log only shows "OS=Linux" before the timeout, which tells us nothing).

My preference: (1) + (3) together. That directly addresses the symptom and removes the re-run tax without hiding real regressions in the probe tools.

## Related

No existing issue found via `gh issue list` search for "Report environment details", "environment details timed out", or "capture-hw-details timeout".

**Comment by kwasd** [comment]
The problem is somewhere in GPU stack leading to clinfo hanging (backtrace will follow).

No point in increasing the timeout, as this will just make the wait longer before failing.
No point in ignoring the error, as further test results cannot be trusted.

Workaround is a watchdog disabling stuck runners until reboot, I am now implementing it.

**Comment by kwasd** [comment]
Watchdog is implemented in-place on affected machines, runs from `cron.hourly`. 
Next step is getting a backtrace and reporting.

**Comment by kwasd** [comment]
Everything is alive so far, maybe due to recent upgrade of the kernel driver. If any runner dies, need to do something like `gdb -p $(pgrep clinfo)` (inside the container, so it could collect more details), then collect the `bt`
@exolyr FYI in case I will be unavailable at the moment.
